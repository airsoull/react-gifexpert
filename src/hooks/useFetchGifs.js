import { useState, useEffect } from 'react';
import getGifs from '../helpers/getGifs';

const useFetchGifs = (category, limit = 9) => {
  const [state, setState] = useState({
    images: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category, limit).then((imgs) => setState({
      images: imgs,
      loading: false,
    }));
  }, [category, limit]);

  return state;
};

export default useFetchGifs;
