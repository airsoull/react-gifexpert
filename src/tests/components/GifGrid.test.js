import React from 'react';
import useFetchGifs from '../../hooks/useFetchGifs';

const { shallow } = require('enzyme');
const { default: GifGrid } = require('../../components/GifGrid');

jest.mock('../../hooks/useFetchGifs');

describe('test for GifGrid component', () => {
  test('should show the component correctly', () => {

    useFetchGifs.mockReturnValue({
      images: [],
      loading: true,
    });

    const category = 'foo-bar';
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are loads', () => {
    const images = [
      {
        id: 1,
        title: 'foo-bar',
        url: 'http://www.example.com/img.jpg',
      },
      {
        id: 2,
        title: 'foo-bar',
        url: 'http://www.example.com/img.jpg',
      },
    ];

    useFetchGifs.mockReturnValue({
      images,
      loading: false,
    });

    const category = 'foo-bar';
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(images.length);
  });
});
