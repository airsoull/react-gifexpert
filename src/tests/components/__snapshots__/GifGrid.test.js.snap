// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for GifGrid component should show items when images are loads 1`] = `
<Fragment>
  <h3
    className="mt-40"
  >
    foo-bar
  </h3>
  <div
    className="card-grid animate__animated animate__zoomIn"
  >
    <GifGridItem
      key="1"
      title="foo-bar"
      url="http://www.example.com/img.jpg"
    />
    <GifGridItem
      key="2"
      title="foo-bar"
      url="http://www.example.com/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`test for GifGrid component should show the component correctly 1`] = `
<Fragment>
  <h3
    className="mt-40"
  >
    foo-bar
  </h3>
  <div
    className="card-grid animate__animated animate__zoomIn"
  >
    <p>
      Loading...
    </p>
  </div>
</Fragment>
`;
